### DISCUSSION TITLE
请问mirai-core群匿名消息是否可以获取发送者的QQ号？

### DISCUSSION THREAD

**Discussion by applearound** [mamoe/mirai#1346] [Q&A] [answered]
我发现文档和源码注释里面没有写得很清楚好像，其实有两个问题。
1. 如何确定群消息是匿名消息？（是否只能通过 instanceof 判断）
2. 如果一条消息是匿名消息，发送者的哪些真实信息会被屏蔽？（是否可以获取其id，也就是QQ号）

**Comment by Karlatemp** [comment, marked as answer]
1. Yes, only `instanceof`
2. We don't know (no, but you can get `anonymousId`)

**Reply by applearound** [reply]
I got it. Thank you.

**Reply by applearound** [reply]
As for anonymous member, to get the anonymous character name, which property is correct, nameCard or nick? or something better?

**Reply by Him188** [reply]
I guess it would be `nameCard`.
